pyincore_data/utils/nsiutil.py

```diff
@@ -0,0 +1,247 @@
+            if region == "WestCoast":
+                if no_stories_ <= 3:
+                    sheet = 'LowRise-Pre1950' if year_built_ <= 1950 else (
+                        'LowRise-1950-1970' if year_built_ <= 1970 else 'LowRise-Post1970')
+                elif no_stories_ <= 7:
+                    sheet = 'MidRise-Pre1950' if year_built_ <= 1950 else (
+                        'MidRise-1950-1970' if year_built_ <= 1970 else 'MidRise-Post1970')
+                else:
+                    sheet = 'HighRise-Pre1950' if year_built_ <= 1950 else (
+                        'HighRise-1950-1970' if year_built_ <= 1970 else 'HighRise-Post1970')
+
+                fallback_sheets = [
+                    sheet.replace("HighRise", "MidRise").replace("LowRise", "MidRise"),
+                    sheet.replace("HighRise", "LowRise").replace("MidRise", "LowRise")
+                ]
+            else:  # MidWest & EastCoast only have 3 sheets
+                if no_stories_ <= 3:
+                    sheet = 'LowRise'
+                elif no_stories_ <= 7:
+                    sheet = 'MidRise'
+                else:
+                    sheet = 'HighRise'
+
+                fallback_sheets = ['MidRise', 'LowRise'] if sheet == 'HighRise' else ['LowRise']
+
+            found_match = False
+            for check_sheet in [sheet] + fallback_sheets:
+                if occ_type_ in o2b_dict.get(check_sheet, pd.DataFrame()).index:
+                    row = o2b_dict[check_sheet].loc[occ_type_].dropna()
+                    if not row.empty:
+                        if check_sheet != sheet:
+                            logger.debug(f"'{occ_type_}' not found in sheet '{sheet}'")
+                            exact_match_flag = "No"
+                            logger.debug(f"Applying fallback: '{occ_type_}' found in '{check_sheet}' instead.")
+                            fallback_count += 1
+                        sheet = check_sheet
+                        found_match = True
+                        break
+
+            # **New Check: If still not found, check LowRise for MidWest & EastCoast**
+            if not found_match and region in ["MidWest", "EastCoast"]:
+                if 'LowRise' in o2b_dict and occ_type_ in o2b_dict['LowRise'].index:
+                    row = o2b_dict['LowRise'].loc[occ_type_].dropna()
+                    if not row.empty:
+                        logger.debug(f"Final fallback: '{occ_type_}' found in 'LowRise'")
+                        sheet = 'LowRise'
+                        fallback_count += 1
+                        found_match = True
+
+            if not found_match:
+                logger.warning(f"'{occ_type_}' not found in any applicable sheet.")
+                struct_typ.append(np.nan)
+                no_stories.append(np.nan)
+                year_built.append(np.nan)
+                dgn_lvl.append(np.nan)
+                exact_match.append("No")
+                cnt_nan += 1
+                continue
```

Comment:
same, this continue will cause skipping the below struct types logic, is that intended?

Reply:
yes